diff --git a/process.d.ts b/process.d.ts
index 63b6525878620b265f9c607edc0864d09d1b93e5..5dfb8d83cd201a10aa4b6a02b215bdbca147703f 100644
--- a/process.d.ts
+++ b/process.d.ts
@@ -256,17 +256,9 @@ declare module "process" {
                 openssl: string;
             }
             type Platform =
-                | "aix"
-                | "android"
                 | "darwin"
-                | "freebsd"
-                | "haiku"
                 | "linux"
-                | "openbsd"
-                | "sunos"
                 | "win32"
-                | "cygwin"
-                | "netbsd";
             type Architecture =
                 | "arm"
                 | "arm64"
@@ -1526,12 +1518,8 @@ declare module "process" {
                  *
                  * Currently possible values are:
                  *
-                 * * `'aix'`
                  * * `'darwin'`
-                 * * `'freebsd'`
                  * * `'linux'`
-                 * * `'openbsd'`
-                 * * `'sunos'`
                  * * `'win32'`
                  *
                  * ```js
@@ -1539,9 +1527,6 @@ declare module "process" {
                  *
                  * console.log(`This platform is ${platform}`);
                  * ```
-                 *
-                 * The value `'android'` may also be returned if the Node.js is built on the
-                 * Android operating system. However, Android support in Node.js [is experimental](https://github.com/nodejs/node/blob/HEAD/BUILDING.md#androidandroid-based-devices-eg-firefox-os).
                  * @since v0.1.16
                  */
                 readonly platform: Platform;
