diff -ruN aria2-1.36.0/src/a2functional.h aria2/src/a2functional.h
--- aria2-1.36.0/src/a2functional.h	2021-08-21 10:28:39.000000000 +0200
+++ aria2/src/a2functional.h	2022-09-09 13:18:33.404636307 +0200
@@ -103,7 +103,7 @@
 }
 
 template <typename T>
-class LeastRecentAccess : public std::binary_function<T, T, bool> {
+class LeastRecentAccess {
 public:
   bool operator()(const std::shared_ptr<T>& lhs,
                   const std::shared_ptr<T>& rhs) const
